### PR TITLE
Use Board Layout names instead of index.

### PR DESCRIPTION
**Pull request by tomprince**

### CODE REVIEW

File: objects/9ad187/script.lua [hunk 1]
```diff
@@ -588,7 +590,7 @@ function difficultyCheck(params)
     if optionalExtraBoard then
         local intNum = math.floor(difficulty / 3) + 2
         difficulty = difficulty + intNum
-        if alternateSetupIndex == 2 then
+        if boardLayout == "Thematic" then
```
Comment (iakona) [inline objects/9ad187/script.lua:593]: just noticed there's a bug here (unrelated to your change), but could you also add a `or param.thematic` here?

File: global-script.lua [hunk 2]
```diff
@@ -2403,9 +2376,9 @@ function MapPlaceCustom()
     local maps = getMapTiles()
     -- board type is guaranteed to either be thematic or normal, and there has to be at least one map tile in the table
     if string.sub(maps[1].getName(),1,4) == "NORM" then
-        alternateSetupIndex = 1
+        boardLayout = "Custom"
```
Comment (iakona) [inline global-script.lua:2379]: what's the idea behind this being `Custom`?

File: objects/9ad187/script.lua (hunk 1 above)
Comment (iakona) [reply objects/9ad187/script.lua:593]: actually nvm, i'll just fix it real quick, noticed a couple other things too

File: global-script.lua (hunk 2 above)
Comment (tomprince) [reply global-script.lua:2379]: This is not one the the standard layouts, so I record that. I'd use that for both normal and  thematic, except things use `Thematic` for calculating difficulty.

File: objects/9ad187/script.lua [hunk 3]
```diff
@@ -278,15 +275,28 @@ function toggleNumPlayers(_, value)
 end
 function updateBoardLayouts(numPlayers)
```
Comment (iakona) [inline objects/9ad187/script.lua:276]: should this function just accept `numBoards` instead now?

Comment (tomprince) [reply objects/9ad187/script.lua:276]: I don't think so. There are a couple of places that call this, and they'd both need to be updated to pass the new value in.

**Review by iakona — APPROVED** [review]

**Pull request merged**